### PR TITLE
gpl: Fix for trivial designs (Replace diverged at initial iteration)

### PR DESCRIPTION
**Pull request by vijayank88**
Rebased with latest commit. Detailed discussion available in #2370

### CODE REVIEW

**Comment by vijayank88** [comment]
@vvbandeira 
Can you help me to resolve `Format Code with pre commit trigger / format (pull_request) ` error.?

**Comment by maliberty** [comment]
Needs formatting

**Comment by maliberty** [comment]
Fails multiple tests:
image

**Comment by vijayank88** [comment]
> Needs formatting

I am 1st time updating with source file `.cpp`. 
Please guide me on how to format this.

**Comment by vijayank88** [comment]
> Fails multiple tests: image

Unit test passes. Need update in `def.ok` file to pass. I will update once, formatting done.

**Comment by maliberty** [comment]
> > Needs formatting
> 
> I am 1st time updating with source file `.cpp`. Please guide me on how to format this.

clang-format -i <file>

**Comment by vijayank88** [comment]
@maliberty @vvbandeira 
`test/regression` produces different logs in `CI` vs `local` run.
Left side CI logs differ from logs with local test on right side:
ci_local_regression_diff

How to resolve this?

File: src/gpl/test/simple09.ok [hunk 1]
```diff
@@ -39,6 +39,6 @@
 [INFO GPL-0028] BinCnt: 16 16
 [INFO GPL-0029] BinSize: 3872 3850
 [INFO GPL-0030] NumBins: 256
-[NesterovSolve] Iter: 1 overflow: 0 HPWL: 10073
+[NesterovSolve] Iter: 1 overflow: 0 HPWL: 116507
```
Comment (maliberty) [inline src/gpl/test/simple09.ok:42]: This change doesn't work well and this is the symptom.   The isnan() code in NesterovPlace::init does not fire.  I think you are better off with something like:
```
@@ -540,7 +540,9 @@ int NesterovPlace::doNesterovPlace(int start_iter)
 
       debugPrint(log_, GPL, "np", 1, "NewStepLength: {:g}", newStepLength);
 
-      if (isnan(newStepLength) || isinf(newStepLength)) {
+      if (isinf(newStepLength)) {
+        newStepLength = 0;
+      } else if (isnan(newStepLength)) {
         isDiverged_ = true;
         divergeMsg_ = "RePlAce diverged at newStepLength.";
         divergeCode_ = 305;
```

Comment (vijayank88) [reply src/gpl/test/simple09.ok:42]: I tried to implement above changes, but unable to compile it. Error follows:
```
/home/vijayan/OPENROAD_FLOW/OpenROAD_FLOW/OpenROAD/src/gpl/src/nesterovPlace.cpp: In member function \u2018int gpl::NesterovPlace::doNesterovPlace(int)\u2019:
/home/vijayan/OPENROAD_FLOW/OpenROAD_FLOW/OpenROAD/src/gpl/src/nesterovPlace.cpp:544:11: error: \u2018isinif\u2019 was not declared in this scope
       if (isinif(newStepLength)) {
           ^~~~~~
/home/vijayan/OPENROAD_FLOW/OpenROAD_FLOW/OpenROAD/src/gpl/src/nesterovPlace.cpp:544:11: note: suggested alternative: \u2018isinff\u2019
       if (isinif(newStepLength)) {
           ^~~~~~
           isinff
gmake[2]: *** [src/gpl/CMakeFiles/gpl.dir/build.make:159: src/gpl/CMakeFiles/gpl.dir/src/nesterovPlace.cpp.o] Error 1
```

Comment (vijayank88) [reply src/gpl/test/simple09.ok:42]: I have updated with above format, but still got same logs.

File: src/gpl/src/nesterovPlace.cpp [hunk 2]
```diff
@@ -875,6 +877,9 @@ float NesterovPlace::getStepLength(
   debugPrint(
       log_, GPL, "getStepLength", 1, "GradientDistance: {:g}", gradDistance);
 
+  if (gradDistance == 0) {
+    return 0;
+  }
```
Comment (maliberty) [inline src/gpl/src/nesterovPlace.cpp:882]: You are still blocking the isinf call by having this code.

Comment (vijayank88) [reply src/gpl/src/nesterovPlace.cpp:882]: shall I remove that if block?

Comment (maliberty) [reply src/gpl/src/nesterovPlace.cpp:882]: yes

Comment (vijayank88) [reply src/gpl/src/nesterovPlace.cpp:882]: If I remove `if` block, not resolving this [testcase](https://github.com/The-OpenROAD-Project/OpenROAD/pull/2370#issuecomment-1286370120)

**Comment by maliberty** [comment]
replaced by #3170

**Comment by maliberty** [comment]
replaced by #3170

**Pull request closed without merging**